src/components/images/imageLoader.js

```diff
@@ -72,7 +72,9 @@ import 'css!./style';
         }
 
         if (userSettings.enableBlurhash()) {
-            if (!target.classList.contains('blurhashed', 'non-blurhashable') && blurhashstr) {
+            if (!target.classList.contains('blurhashed', 'non-blurhashable')
+                && blurhashstr
+                && target.querySelectorAll('canvas').length === 0) {
```

Comment:
I'm not sure exactly why you saw this, but this doesn't make sense, given that we remove the data-blurhash attribute of the elements (so in a further pass it would be impossible to draw the canvas again, as the attribute doesn't exist) and we also add a class that we use to check. This is completely redundant. See lines 27, 48 and 49.

Reply:
To paraphrase my comment on Matrix: I had a weird issue where going to a library page, then to itemDetails then back to the library was causing items to have 3 canvases, one of which had an opacity of 0 and two with an opacity of 1.

Not finding what could cause this, I added that check in order to make sure it'll only create a canvas if there is none.